### PR TITLE
Fix handling of virtual packages

### PR DESCRIPTION
**Pull request by ipa-mdl**
fixes #521

### CODE REVIEW

**Comment by ipa-mdl** [comment]
includes @flixr's fix for #589 

@wjwwood: #591 breaks our tests, they pass if rebased to 1e69f89988e8d2fbe931bc14639eb380bff30844

File: test/test_rosdep_debian.py [hunk 1]
```diff
@@ -64,9 +64,9 @@ def test_read_apt_cache_showpkg():
     m = Mock()
     with open(os.path.join(get_test_dir(), 'showpkg-curl-wget-libcurl-dev'), 'r') as f:
         m.return_value = f.read()
-
-    results = list(_read_apt_cache_showpkg(['curl', 'wget', '_not_existing', 'libcurl-dev'], exec_fn=m))
-    assert len(results) == 4, results
+    pkgs = ['curl', 'wget', '_not_existing', 'libcurl-dev', 'ros-kinetic-rc-genicam-api']
```
Comment (flixr) [inline test/test_rosdep_debian.py]: will this still work as intended after the next kinetic sync, from which on ros-kinetic-rc-genicam-api will be available?

Comment (ipa-mdl) [reply test/test_rosdep_debian.py]: Yes, this just tests the format parser based on the text file.

Comment (flixr) [reply test/test_rosdep_debian.py]: ah, right...

**Comment by ipa-mdl** [comment]
> #591 breaks our tests, they pass if rebased to 1e69f89

My bad, I patched a bug that wasn't there and the unit tests caught this :+1: 

@wjwwood: please review

**Comment by wjwwood** [comment]
Thanks for working to fix the issue guys.

**Pull request merged**